crates/sncast/src/helpers/interactive.rs

```diff
@@ -0,0 +1,114 @@
+use crate::helpers::config::get_global_config_path;
+use anyhow::{Context, Result};
+use camino::Utf8PathBuf;
+use configuration::search_config_upwards_relative_to;
+use dialoguer::theme::ColorfulTheme;
+use dialoguer::Select;
+use std::env::current_dir;
+use std::fs::File;
+use std::io::{Read, Write};
+use std::path::Path;
+use toml_edit::{DocumentMut, Item, Table, Value};
+
+pub fn ask_to_add_as_default(account: &str) -> Result<()> {
+    let mut options = Vec::new();
+
+    if let Ok(global_path) = get_global_config_path() {
+        let option = format!(
+            "Yes, global default account ({}).",
+            to_tilde_path(global_path.as_str())
+        );
+        options.push(option);
+    }
+
+    if let Ok(current_path) = current_dir() {
+        let current_path_utf8 = Utf8PathBuf::from_path_buf(current_path.clone())
+            .expect("Failed to convert current directory to Utf8PathBuf");
+
+        if let Ok(local_path) = search_config_upwards_relative_to(&current_path_utf8) {
+            let option = format!(
+                "Yes, local default account ({}).",
+                to_tilde_path(local_path.as_str())
+            );
+            options.push(option);
+        }
+    }
+
+    options.push("No".to_string());
+
+    let selection = Select::with_theme(&ColorfulTheme::default())
+        .with_prompt("Do you want to make this account default?")
+        .items(&options)
+        .default(0)
+        .interact()
+        .context("Failed to display selection dialog")?;
+
+    match options[selection].as_str() {
```

Comment:
Could we use an enum here somehow? So our logic doesn't depend on matching strings.

Reply:
I don't think so. I can see that it is common way to achieve this: https://github.com/software-mansion/scarb/blob/main/scarb/src/bin/scarb/interactive.rs#L40

Reply:
I see that in Scarb it's implemented this way but I wouldn't consider this a criterion. Imo num fits here well - selection message is defined only once and it looks neater.

Reply:
https://github.com/foundry-rs/starknet-foundry/pull/2792

Reply:
That's how it would have to look like

Reply:
Okay, I'll take a look at this later.